Fix various bugs

kontura opened the pull request:
I tested drpm_make and drpm_read on ~800 pkgs, it should now work more reliably.

There is still some problem with drpm_apply, but I will fix that later as it is not used by createrepo_c. (https://github.com/rpm-software-management/createrepo_c/pull/172)

dmach commented on src/drpm_write.c:
```diff
@@ -117,11 +117,12 @@ int write_deltarpm(struct deltarpm *delta)
                 goto cleanup;
-            for (uint32_t i = 0; i < delta->offadj_elems_count; i += 2) {
+            uint32_t offadj_elems_size = delta->offadj_elems_count * 2;
+            for (uint32_t i = 0; i < offadj_elems_size; i += 2) {
```
It would be good to describe why the odd/even indexes are used.
I haven't seem larger context than what's in the diff, maybe it's explained somewhere near and doesn't have to be documented.

dmach commented on src/drpm_rpm.c:
```diff
@@ -626,11 +626,10 @@ int rpm_get_comp_level(struct rpm *rpmst, unsigned short *level)
 
-    if (strlen(payload_flags) != 1 ||
-        payload_flags[0] < '1' || payload_flags[0] > '9')
+    if (strlen(payload_flags) > 2)
```
comment that this is maximum length of compression level string

dmach commented on src/drpm_make.c:
```diff
@@ -535,7 +535,7 @@ int parse_cpio_from_rpm_filedata(struct rpm *rpm_file,
                 goto cleanup_fail;
             cpio_pos += read_len;
-            if (!S_ISLNK(file.mode) &&
+            if (!S_ISLNK(file.mode) && !skip &&
```
Is it possible to easily explain what `skip` does?
If that's the case, please add a comment.

kontura replied:
I have added a small comment to the condition, `skip` is explained above: https://github.com/rpm-software-management/drpm/blob/master/src/drpm_make.c#L386

kontura commented:
I have added the requested comments and one more commit fixing mentioned bug in drpm_apply. 

(I fixed it by looking at how the conditions are set up in [applydeltarpm.c](https://github.com/rpm-software-management/deltarpm/blob/master/applydeltarpm.c#L882) because it uses exactly the same logic. However I don't understand it enough to provide helpful comments on why they should be this way, if necessary I can spend some more time to understand it and add the comments)

Conan-Kudo commented on src/drpm_make.c:
```diff
@@ -535,7 +535,8 @@ int parse_cpio_from_rpm_filedata(struct rpm *rpm_file,
                 goto cleanup_fail;
             cpio_pos += read_len;
-            if (!S_ISLNK(file.mode) &&
+            /* Do not add file data of skipped files or simlinks to cpio  */
```
`simlinks` -> `symlinks`

kontura replied:
Fixed it, thank you.

Conan-Kudo commented:
@kontura Have you tested with F30 and F31/Rawhide trees to be sure it works with different payloads properly?

kontura commented:
@Conan-Kudo I tested with F31/Rawhide and RHEL8 (which I think has payloads like F30) that is packages with zstd and xz compression.
